File: app/src/main/java/com/prisyazhnuy/newsapp/data/local/db/NewsDAO.java
package com.prisyazhnuy.newsapp.data.local.db;

import com.prisyazhnuy.newsapp.data.pojo.Article;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;

/**
 * max.pr on 04.03.2018.
 */

public interface NewsDAO {

    Completable insert(Article article);

    Completable delete(long id);

    Observable<List<Article>> getAll();
}

File: app/src/main/java/com/prisyazhnuy/newsapp/dataKotlin/sources/db/dao/NewsDAO.kt
package com.prisyazhnuy.newsapp.dataKotlin.sources.db.dao

import android.arch.persistence.room.*
import com.prisyazhnuy.newsapp.dataKotlin.entity.News
import io.reactivex.Flowable
import io.reactivex.Observable

/**
 * max.pr on 29.03.2018.
 */
@Dao
interface NewsDAO {
    @Query("SELECT * FROM news")
    fun loadAllNews(): Flowable<List<News>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(news: News)

    @Delete
    fun delete(news: News)
}